frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionDetailsHeaderActions.tsx

```diff
@@ -53,15 +66,79 @@ const ModelVersionsDetailsHeaderActions: React.FC<ModelVersionsDetailsHeaderActi
 
   const isFineTuningEnabled = useIsAreaAvailable(SupportedArea.FINE_TUNING).status;
 
-  const { tuningData, loaded, loadError } = useModelVersionTuningData(
-    isLabTuneModalOpen ? mv.id : null,
-    mv,
-    registeredModel,
-  );
-  const [modelArtifacts] = useModelArtifactsByVersionId(mv.id);
-  const isOciModel = isOciModelUri(modelArtifacts.items.map((artifact) => artifact.uri)[0]);
+  const { inputModelLocationUri, modelRegistryDisplayName, outputModelRegistryApiUrl } =
+    React.useMemo(() => {
+      const registryService = modelRegistryServices.find(
+        (s) => s.metadata.name === preferredModelRegistry?.metadata.name,
+      );
+      return {
+        inputModelLocationUri: modelArtifacts.items[0]?.uri,
+        modelRegistryDisplayName: registryService
+          ? getDisplayNameFromK8sResource(registryService)
+          : '',
```

Comment:
It's kind of a shame to lose the encapsulation we had from the `useModelVersionTuningData` hook here, and we've now duplicated all the logic from it because it is still used elsewhere (in ModelVersionsTableRow). Can we instead change `useModelVersionTuningData` into a util `getModelVersionTuningData` that takes the artifacts as a param, and then use that here instead of duplicating its logic? We'll have to also replace the other usage of it in ModelVersionsTableRow so it relies on the useModelArtifactsByVersionId call that it already makes outside its usage of useModelVersionTuningData (looks like we had a double fetch there as well). We'll have to just pull the loaded, loadError out of that existing useModelArtifactsByVersionId call instead (the one on line 54 in ModelVersionsTableRow).